File: Tools/update_version.js
const fs = require('fs');
const path = require('path');
const { exec } = require('child_process');
// Set the file path
const filePath = path.join(__dirname,'../projects','dynamic-form','package.json');

// Read the file content
fs.readFile(filePath, 'utf8', (err, data) => {
    if (err) {
        console.error('Error reading the file:', err);
        return;
    }

    // Parse the JSON data
    let json;
    try {
        json = JSON.parse(data);
    } catch (e) {
        console.error('Error parsing JSON:', e);
        return;
    }

    // Get the version string
    let version = json.version;
    if (!version) {
        console.error('Version string not found in JSON');
        return;
    }

    // Split the version string into major, minor, and patch
    let [major, minor, patch] = version.split('.').map(Number);

    // Increment the version
    patch += 1;
    if (patch > 99) {
        patch = 0;
        minor += 1;
    }
    if (minor > 9) {
        minor = 0;
        major += 1;
    }

    // Create the new version string
    const newVersion = `${major}.${minor}.${patch}`;

    // Update the JSON with the new version
    json.version = newVersion;

    // Write the updated JSON back to the file
    fs.writeFile(filePath, JSON.stringify(json, null, 2), 'utf8', err => {
        if (err) {
            console.error('Error writing to the file:', err);
            return;
        }
        console.log('Version updated successfully:');
        console.log(`Old Version: ${version}`);
        console.log(`New Version: ${newVersion}`);
    });
});

File: Tools/update_version.py
import json
from pathlib import Path

lib_name='dynamic-form'

tools_dir = Path(__file__).resolve().parent
main_dir = tools_dir.parent
project_dir = main_dir / 'projects' / lib_name
def update_version():
    # Read the file content
    try:
        with open(project_dir / 'package.json', 'r', encoding='utf8') as file:
            data = file.read()
    except IOError as e:
        print(f"Error reading the file: {e}")
        exit()

    # Parse the JSON data
    try:
        json_data = json.loads(data)
    except json.JSONDecodeError as e:
        print(f"Error parsing JSON: {e}")
        exit()

    # Get the version string
    version = json_data.get('version')
    if not version:
        print("Version string not found in JSON")
        exit()

    # Split the version string into major, minor, and patch
    major, minor, patch = map(int, version.split('.'))

    # Increment the version
    patch += 1
    if patch > 99:
        patch = 0
        minor += 1
    if minor > 9:
        minor = 0
        major += 1

    # Create the new version string
    new_version = f"{major}.{minor}.{patch}"

    # Update the JSON with the new version
    json_data['version'] = new_version

    # Write the updated JSON back to the file
    try:
        with open(project_dir / 'package.json', 'w', encoding='utf8') as file:
            json.dump(json_data, file, indent=2)
        print("Version updated successfully:")
        print(f"Old Version: {version}")
        print(f"New Version: {new_version}")
    except IOError as e:
        print(f"Error writing to the file: {e}")
